From 9d0ab3aa9d0dd1a3d3885b44448acb262396a57d Mon Sep 17 00:00:00 2001
From: Sloane Hertel <19572925+s-hertel@users.noreply.github.com>
Date: Thu, 30 Jun 2022 13:19:44 -0400
Subject: [PATCH 2/2] Make unit test for missing git executable more generic
 (#78173)

* Make unit test for missing git executable more generic

* use MagicMock side_effect to raise exception instead
---
 test/units/galaxy/test_collection_install.py | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/test/units/galaxy/test_collection_install.py b/test/units/galaxy/test_collection_install.py
index 7525a194ca..d83fe42054 100644
--- a/test/units/galaxy/test_collection_install.py
+++ b/test/units/galaxy/test_collection_install.py
@@ -181,13 +181,14 @@ def test_concrete_artifact_manager_scm_no_executable(monkeypatch):
     monkeypatch.setattr(collection.concrete_artifact_manager.subprocess, 'check_call', mock_subprocess_check_call)
     mock_mkdtemp = MagicMock(return_value='')
     monkeypatch.setattr(collection.concrete_artifact_manager, 'mkdtemp', mock_mkdtemp)
+    mock_get_bin_path = MagicMock(side_effect=[ValueError('Failed to find required executable')])
+    monkeypatch.setattr(collection.concrete_artifact_manager, 'get_bin_path', mock_get_bin_path)
 
     error = re.escape(
         "Could not find git executable to extract the collection from the Git repository `https://github.com/org/repo`"
     )
-    with mock.patch.dict(os.environ, {"PATH": ""}):
-        with pytest.raises(AnsibleError, match=error):
-            collection.concrete_artifact_manager._extract_collection_from_git(url, version, b'path')
+    with pytest.raises(AnsibleError, match=error):
+        collection.concrete_artifact_manager._extract_collection_from_git(url, version, b'path')
 
 
 @pytest.mark.parametrize(
-- 
2.36.1

